### ISSUE TITLE
Timeout warning/error

### ISSUE THREAD

**Issue by dittops** [huggingface/datatrove#75]
I'm trying to run the common crawl example. I have replaced the executor with Local executor but I'm getting timeout error and warning. 

```
2024-01-31 15:31:30.307 | WARNING  | datatrove.pipeline.extractors.base:timeout_extract:49 - ⏰ Timeout while cleaning record text. Skipping record.
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fb2d78c8dc0>
Traceback (most recent call last):
  File "/usr/lib/python3.10/weakref.py", line 106, in remove
    def remove(wr, selfref=ref(self), _atomic_removal=_remove_dead_weakref):
  File "/home/developer/datatrove/src/datatrove/pipeline/extractors/base.py", line 41, in signal_handler
    raise TimeoutError
TimeoutError: 
2024-01-31 15:31:30.942 | WARNING  | datatrove.pipeline.extractors.base:timeout_extract:49 - ⏰ Timeout while cleaning record text. Skipping record.
2024-01-31 15:31:31.703 | WARNING  | datatrove.pipeline.extractors.base:timeout_extract:49 - ⏰ Timeout while cleaning record text. Skipping record.
2024-01-31 15:31:31.905 | WARNING  | datatrove.pipeline.extractors.base:timeout_extract:49 - ⏰ Timeout while cleaning record text. Skipping record.
2024-01-31 15:31:33.132 | WARNING  | datatrove.pipeline.extractors.base:timeout_extract:49 - ⏰ Timeout while cleaning record text. Skipping record.
2024-01-31 15:31:33.355 | WARNING  | datatrove.pipeline.extractors.base:timeout_extract:49 - ⏰ Timeout while cleaning record text. Skipping record.
2024-01-31 15:31:33.576 | WARNING  | datatrove.pipeline.extractors.base:timeout_extract:49 - ⏰ Timeout while cleaning record text. Skipping record.
2024-01-31 15:31:34.147 | WARNING  | datatrove.pipeline.extractors.base:timeout_extract:49 - ⏰ Timeout while cleaning record text. Skipping record.
2024-01-31 15:31:34.574 | WARNING  | datatrove.pipeline.extractors.base:timeout_extract:49 - ⏰ Timeout while cleaning record text. Skipping record.
2024-01-31 15:31:35.368 | WARNING  | datatrove.pipeline.extractors.base:timeout_extract:49 - ⏰ Timeout while cleaning record text. Skipping record.
2024-01-31 15:31:36.850 | ERROR    | datatrove.executor.base:_run_for_rank:74 - 
Traceback (most recent call last):

  File "<string>", line 1, in <module>
  File "/home/developer/datatrove/.env/lib/python3.10/site-packages/multiprocess/forkserver.py", line 273, in main
    code = _serve_one(child_r, fds,
           │          │        └ [49, 50, 51, 52, 53, 54]
           │          └ 8
           └ <function _serve_one at 0x7fb351575ab0>
  File "/home/developer/datatrove/.env/lib/python3.10/site-packages/multiprocess/forkserver.py", line 312, in _serve_one
    code = spawn._main(child_r, parent_sentinel)
           │     │     │        └ 4
           │     │     └ 8
           │     └ <function _main at 0x7fb351574dc0>
           └ <module 'multiprocess.spawn' from '/home/developer/datatrove/.env/lib/python3.10/site-packages/multiprocess/spawn.py'>
  File "/home/developer/datatrove/.env/lib/python3.10/site-packages/multiprocess/spawn.py", line 129, in _main
    return self._bootstrap(parent_sentinel)
           │    │          └ 4
           │    └ <function BaseProcess._bootstrap at 0x7fb351821ab0>
           └ <ForkServerProcess name='ForkServerPoolWorker-39' parent=283031 started daemon>
  File "/home/developer/datatrove/.env/lib/python3.10/site-packages/multiprocess/process.py", line 314, in _bootstrap
    self.run()
    │    └ <function BaseProcess.run at 0x7fb351821120>
    └ <ForkServerProcess name='ForkServerPoolWorker-39' parent=283031 started daemon>
  File "/home/developer/datatrove/.env/lib/python3.10/site-packages/multiprocess/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
    │    │        │    │        │    └ {}
    │    │        │    │        └ <ForkServerProcess name='ForkServerPoolWorker-39' parent=283031 started daemon>
    │    │        │    └ (<multiprocess.queues.SimpleQueue object at 0x7fb35155a2c0>, <multiprocess.queues.SimpleQueue object at 0x7fb2d78a3340>, None...
    │    │        └ <ForkServerProcess name='ForkServerPoolWorker-39' parent=283031 started daemon>
    │    └ <function worker at 0x7fb2d78b0700>
    └ <ForkServerProcess name='ForkServerPoolWorker-39' parent=283031 started daemon>
  File "/home/developer/datatrove/.env/lib/python3.10/site-packages/multiprocess/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
                    │     │       └ {}
                    │     └ (29,)
                    └ functools.partial(<bound method LocalPipelineExecutor._launch_run_for_rank of <datatrove.executor.local.LocalPipelineExecutor...

  File "/home/developer/datatrove/src/datatrove/executor/local.py", line 46, in _launch_run_for_rank
    return self._run_for_rank(rank, local_rank)
           │    │             │     └ 29
           │    │             └ 29
           │    └ <function PipelineExecutor._run_for_rank at 0x7fb350aefac0>
           └ <datatrove.executor.local.LocalPipelineExecutor object at 0x7fb2d78a3520>

> File "/home/developer/datatrove/src/datatrove/executor/base.py", line 62, in _run_for_rank
    deque(pipelined_data, maxlen=0)
    │     └ <generator object DiskWriter.run at 0x7fb2d5fa12a0>
    └ <class 'collections.deque'>

  File "/home/developer/datatrove/src/datatrove/pipeline/writers/disk_base.py", line 51, in run
    for document in data:
        │           └ <generator object BaseFilter.run at 0x7fb2d5fa1540>
        └ Document(text='|\n|\n2019 cheap ray ban sunglasses online 2019, Tips on Buying Ray Ban Sunglasses Overstock There are so many...

  File "/home/developer/datatrove/src/datatrove/pipeline/filters/base_filter.py", line 46, in run
    for doc in data:
        │      └ <generator object BaseFilter.run at 0x7fb2d5fa1ee0>
        └ Document(text='|\n|\n2019 cheap ray ban sunglasses online 2019, Tips on Buying Ray Ban Sunglasses Overstock There are so many...

  File "/home/developer/datatrove/src/datatrove/pipeline/filters/base_filter.py", line 46, in run
    for doc in data:
        │      └ <generator object BaseFilter.run at 0x7fb2d62b28f0>
        └ Document(text="© 2023 Barchart.com, Inc. All market data is hosted and powered by Barchart.\nInformation presented is provide...

  File "/home/developer/datatrove/src/datatrove/pipeline/filters/base_filter.py", line 46, in run
    for doc in data:
        │      └ <generator object BaseFilter.run at 0x7fb2d66a5230>
        └ Document(text="© 2023 Barchart.com, Inc. All market data is hosted and powered by Barchart.\nInformation presented is provide...
  [Previous line repeated 1 more time]

  File "/home/developer/datatrove/src/datatrove/pipeline/extractors/base.py", line 61, in run
    doc.text = self.timeout_extract(doc)
    │   │      │    │               └ Document(text='<!DOCTYPE html>\n<html>\n\t<head>\n\t\t<meta charset="utf-8">\n<meta http-equiv="X-UA-Compatible" content="IE=...
    │   │      │    └ <function BaseExtractor.timeout_extract at 0x7fb350b111b0>
    │   │      └ 🛢 - EXTRAC: ⛏ Trafilatura
    │   └ '<!DOCTYPE html>\n<html>\n\t<head>\n\t\t<meta charset="utf-8">\n<meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"...
    └ Document(text='<!DOCTYPE html>\n<html>\n\t<head>\n\t\t<meta charset="utf-8">\n<meta http-equiv="X-UA-Compatible" content="IE=...

  File "/home/developer/datatrove/src/datatrove/pipeline/extractors/base.py", line 55, in timeout_extract
    signal.setitimer(signal.ITIMER_REAL, 0)
    │      │         │      └ 0
    │      │         └ <module 'signal' from '/usr/lib/python3.10/signal.py'>
    │      └ <built-in function setitimer>
    └ <module 'signal' from '/usr/lib/python3.10/signal.py'>

  File "/home/developer/datatrove/src/datatrove/pipeline/extractors/base.py", line 41, in signal_handler
    raise TimeoutError

TimeoutError
2024-01-31 15:31:37.033 | INFO     | datatrove.executor.local:_launch_run_for_rank:51 - 1/48 tasks completed.
multiprocess.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/home/developer/datatrove/.env/lib/python3.10/site-packages/multiprocess/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/home/developer/datatrove/src/datatrove/executor/local.py", line 46, in _launch_run_for_rank
    return self._run_for_rank(rank, local_rank)
  File "/home/developer/datatrove/src/datatrove/executor/base.py", line 75, in _run_for_rank
    raise e
  File "/home/developer/datatrove/src/datatrove/executor/base.py", line 62, in _run_for_rank
    deque(pipelined_data, maxlen=0)
  File "/home/developer/datatrove/src/datatrove/pipeline/writers/disk_base.py", line 51, in run
    for document in data:
  File "/home/developer/datatrove/src/datatrove/pipeline/filters/base_filter.py", line 46, in run
    for doc in data:
  File "/home/developer/datatrove/src/datatrove/pipeline/filters/base_filter.py", line 46, in run
    for doc in data:
  File "/home/developer/datatrove/src/datatrove/pipeline/filters/base_filter.py", line 46, in run
    for doc in data:
  [Previous line repeated 1 more time]
  File "/home/developer/datatrove/src/datatrove/pipeline/extractors/base.py", line 61, in run
    doc.text = self.timeout_extract(doc)
  File "/home/developer/datatrove/src/datatrove/pipeline/extractors/base.py", line 55, in timeout_extract
    signal.setitimer(signal.ITIMER_REAL, 0)
  File "/home/developer/datatrove/src/datatrove/pipeline/extractors/base.py", line 41, in signal_handler
    raise TimeoutError
TimeoutError
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/developer/datatrove/test.py", line 59, in <module>
    run_example()
  File "/home/developer/datatrove/test.py", line 56, in run_example
    print(executor.run())
  File "/home/developer/datatrove/src/datatrove/executor/local.py", line 80, in run
    stats = list(
  File "/home/developer/datatrove/.env/lib/python3.10/site-packages/multiprocess/pool.py", line 873, in next
    raise value
TimeoutError
```

The code i'm trying with

```
import sys

from datatrove.executor.slurm import SlurmPipelineExecutor
from datatrove.executor.local import LocalPipelineExecutor
from datatrove.pipeline.extractors import Trafilatura
from datatrove.pipeline.filters import (
    GopherQualityFilter,
    GopherRepetitionFilter,
    LanguageFilter,
    ListFilter,
    URLFilter,
)
from datatrove.pipeline.readers import WarcReader
from datatrove.pipeline.writers.jsonl import JsonlWriter


# DUMP should be given as an argument. Example: CC-MAIN-2023-23
if len(sys.argv) != 2:
    print("Argument required: dump name")
    sys.exit(-1)
DUMP = sys.argv[1]

MAIN_OUTPUT_PATH = "s3://commoncrawl-processed"

def run_example():
    executor = LocalPipelineExecutor(
        # job_name=f"cc_{DUMP}",
        pipeline=[
            WarcReader(
                f"s3://commoncrawl/crawl-data/{DUMP}/segments/",
                glob_pattern="*/warc/*",  # we want the warc files
                default_metadata={"dump": DUMP},
            ),
            URLFilter(exclusion_writer=JsonlWriter(f"{MAIN_OUTPUT_PATH}/removed/url/{DUMP}")),
            Trafilatura(favour_precision=True),
            LanguageFilter(
                exclusion_writer=JsonlWriter(
                    f"{MAIN_OUTPUT_PATH}/non_english/",
                    output_filename="${language}/" + DUMP + "/${rank}.jsonl.gz",  # folder structure: language/dump/file
                )
            ),
            GopherRepetitionFilter(exclusion_writer=JsonlWriter(f"{MAIN_OUTPUT_PATH}/removed/repetitive/{DUMP}")),
            GopherQualityFilter(exclusion_writer=JsonlWriter(f"{MAIN_OUTPUT_PATH}/removed/quality/{DUMP}")),
            ListFilter(exclusion_writer=JsonlWriter(f"{MAIN_OUTPUT_PATH}/removed/list/{DUMP}")),
            JsonlWriter(f"{MAIN_OUTPUT_PATH}/output/{DUMP}"),
        ],
        workers=48,
        tasks=48,
        logging_dir=f"{MAIN_OUTPUT_PATH}/logs/base_processing/{DUMP}",

    )
    print(executor.run())

if __name__ == "__main__":
    run_example()

```

**Comment by guipenedo** [comment]
I was able to reproduce but haven't yet had the time to further investigate

**Comment by guipenedo** [comment]
Should be (finally) fixed after commit 5015a4c
